From: Felix Fietkau <nbd@nbd.name>
Date: Fri, 28 Oct 2022 18:16:03 +0200
Subject: [PATCH] net: dsa: tag_mtk: assign per-port queues

Keeps traffic sent to the switch within link speed limits

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---

--- a/net/dsa/tag_mtk.c
+++ b/net/dsa/tag_mtk.c
@@ -33,6 +33,8 @@ static struct sk_buff *mtk_tag_xmit(stru
 	if (__skb_put_padto(skb, ETH_ZLEN + MTK_HDR_LEN, false))
 		return NULL;
 
+	skb_set_queue_mapping(skb, dp->index);
+
 	/* Build the special tag after the MAC Source Address. If VLAN header
 	 * is present, it's required that VLAN header and special tag is
 	 * being combined. Only in this way we can allow the switch can parse
